src/System.Net.HttpListener/tests/HttpListenerResponseTests.Headers.cs

```diff
@@ -0,0 +1,817 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.Tests
+{
+    public class HttpListenerResponseHeadersTests : HttpListenerResponseTestBase
+    {
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotOneCoreUAP))]
+        public async Task ContentEncoding_SetCustom_DoesNothing()
+        {
+            // Setting HttpListenerResponse.ContentEncoding does nothing - it is never used.
+            HttpListenerResponse response = await GetResponse();
+            Assert.Null(response.ContentEncoding);
+
+            response.ContentEncoding = Encoding.Unicode;
+            Assert.Equal(Encoding.Unicode, response.ContentEncoding);
+            response.Close();
+
+            string clientResponse = GetClientResponse();
+            Assert.DoesNotContain("Content-Encoding", clientResponse);
+        }
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotOneCoreUAP))]
+        public async Task ContentEncoding_SetDisposed_DoesNothing()
+        {
+            HttpListenerResponse response = await GetResponse();
+            response.Close();
+
+            response.ContentEncoding = Encoding.Unicode;
+            Assert.Equal(Encoding.Unicode, response.ContentEncoding);
+
+            string clientResponse = GetClientResponse();
+            Assert.DoesNotContain("Content-Encoding", clientResponse);
+        }
+
+        [ConditionalFact(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotOneCoreUAP))]
+        public async Task ContentEncoding_SetAfterHeadersSent_DoesNothing()
+        {
+            using (HttpListenerResponse response = await GetResponse())
+            {
+                response.ContentLength64 = SimpleMessage.Length;
+                response.OutputStream.Write(SimpleMessage, 0, SimpleMessage.Length);
+
+                response.ContentEncoding = Encoding.Unicode;
+                Assert.Equal(Encoding.Unicode, response.ContentEncoding);
+            }
+
+            string clientResponse = GetClientResponse();
+            Assert.DoesNotContain("Content-Encoding", clientResponse);
+        }
+
+        [ConditionalTheory(nameof(PlatformDetection) + "." + nameof(PlatformDetection.IsNotOneCoreUAP))]
+        [InlineData("application/json")]
+        [InlineData("  applICATion/jSOn   ")]
+        [InlineData("garbage")]
+        // The managed implementation should set ContentType directly in the headers instead of tracking it with its own field.
+        // The managed implementation should trim the value.
+        [ActiveIssue(18128, TestPlatforms.AnyUnix)]
+        public async Task ContentType_SetAndSend_Success(string contentType)
```

Comment:
@hughbe Would it be better to file separate issues for identified similar compat differences? In this case filing a separate bug saying that response headers require additional validation and stored in place. So that work can be done incrementally and also distributed in case you are otherwise occupied. Linking all to a single issue doesn't give a clear picture of how much work is still left before one can call the umbrella issue done.